tests/scripts/run_demos.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python3
+"""Run the Mbed TLS demo scripts.
+"""
+import glob
+import subprocess
+import sys
+
+def run_demo(demo):
+    """Run the specified demo script. Return True if it succeeds."""
+    returncode = subprocess.call([demo])
+    return returncode == 0
+
+def run_demos(demos):
+    """Run the specified demos and print summary information about failures.
+
+    Return True if all demos passed and False if a demo fails.
+    """
+    failures = []
+    for demo in demos:
+        print('#### {} ####'.format(demo))
+        if not run_demo(demo):
+            failures.append(demo)
+            print('{}: FAIL'.format(demo))
+        print('')
+    successes = len(demos) - len(failures)
+    print('{}/{} demos passed'.format(successes, len(demos)))
+    if failures:
+        print('Failures:', *failures)
+    return not failures
+
+def run_all_demos():
+    """Run all the available demos.
+
+    Return True if all demos passed and False if a demo fails.
+    """
+    all_demos = glob.glob('programs/*/*_demo.sh')
```

Comment:
Only works if run from the project's root. Otherwise the script will run 0 demos and return success. I think this script should work from any location if possible, and in any case complain when it doesn't find any demos to be run.

Reply:
All of our maintenance and test scripts must be run from the project root. I don't see why this one should be different. Running individual demos from a different directory is something a user might do. But why would a user run `run_demos.py`? That's only meant for testing.

Reply:
Ok, I withdraw "should work from any location" but I maintain "should complain otherwise" which I think many of our scripts do already. (In practice, I think testing for "found 0 demos to run" would be a good test.)